CodeSniffer/Standards/Squiz/Sniffs/PHP/LowercasePHPFunctionsSniff.php

```diff
@@ -119,4 +118,22 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
     }//end process()
 
 
+    /**
+     * Determine whether the functionName in question is a built-in
+     * (or extension-provided) function.
+     *
+     * @param string $functionName Name of the function to check
+     *
+     * @return bool
+     */
+    protected function isBuiltInFunction($functionName)
```

Comment:
Thanks for the tips, will do! Think the array hash map should be a static
variable, or not necessary because there will only ever be one instance?
On May 17, 2016 3:50 AM, "Greg Sherwood" <notifications@github.com> wrote:

> In CodeSniffer/Standards/Squiz/Sniffs/PHP/LowercasePHPFunctionsSniff.php
> <https://github.com/squizlabs/PHP_CodeSniffer/pull/1006#discussion_r63477588>
> :
>
> > @@ -119,4 +118,22 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
> >      }//end process()
> >
> >
> > +    /**
> > +     * Determine whether the functionName in question is a built-in
> > +     * (or extension-provided) function.
> > +     *
> > +     * @param string $functionName Name of the function to check
> > +     *
> > +     * @return bool
> > +     */
> > +    protected function isBuiltInFunction($functionName)
>
> I like the above ideas. I'll add one more. Don't make the call an
> additional function. It's cleaner code, but the additional function call
> for every T_STRING token that appears to be a function name actually adds
> significant overhead when checking large codebases. I know, it's horrible.
> But if it shaves a few seconds off a run, it's good.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/squizlabs/PHP_CodeSniffer/pull/1006/files/5399cf4a0c8320f42a4472a2e62953d3b342c1e4#r63477588>
>